selftests/unit/test_cartesian_config.py

```diff
@@ -387,9 +387,11 @@ def testVariableAssignment(self):
                     var += a
                     var <= b
                     system = 2
-                    ddd = ${tests + str(int(system) + 3)}4
-                    error = ${tests + str(system + 3)}4
-                    s.* ?= ${tests + "ahoj"}4
+                    variable-name-with-dashes = sampletext
+                    ddd = tests variant is ${tests}
+                    dashes = show ${variable-name-with-dashes}
```

Comment:
Additionally, I'd suggest adding the following tests as well.
```
1st = 1
starts_with_number = index ${1st}

not_a_substitution = ${}
```

Reply:
This comment is not resolved but with the LGTM on @luckyh's side I understand it as such and won't repeat any feedback already provided.